server/sitemap/sitemap.service.ts

```diff
@@ -0,0 +1,83 @@
+import { Injectable, Logger } from "@nestjs/common";
+import { SitemapStream, streamToPromise } from "sitemap";
+import { PersonalityService } from "../personality/personality.service";
+import { ClaimService } from "../claim/claim.service";
+import { ClaimReviewService } from "../claim-review/claim-review.service";
+const axios = require("axios");
+
+@Injectable()
+export class SitemapService {
+    constructor(
+        private personalityService: PersonalityService,
+        private claimService: ClaimService,
+        private claimReviewService: ClaimReviewService
+    ) {}
+    private readonly logger = new Logger("SitemapService");
+
+    async getSitemap(hostname) {
+        const sites: any[] = [
+            { url: "/" },
+            { url: "/home" },
+            { url: "/about" },
+            { url: "/privacy-policy" },
+            { url: "/code-of-conduct" },
+            { url: "/login" },
+            { url: "/profile" },
+            { url: "/personality" },
+        ];
+
+        const personalities = await this.personalityService.listAll(
```

Comment:
This will eventually be very slow. We should have a loop that paginates over the results.

For now, you can add a TODO and we resolve in follow-up PR.